pwiz_tools/Skyline/Model/Irt/IrtDb.cs

```diff
@@ -527,31 +524,63 @@ public IrtDb SetDocumentXml(SrmDocument doc, string oldXml)
                     rowExists = Convert.ToInt32(cmd.ExecuteScalar()) > 0;
                 }
 
+                using (var cmd = session.Connection.CreateCommand())
+                {
+                    cmd.CommandText = !rowExists
+                        ? @"INSERT INTO DocumentXml (Xml) VALUES (?)"
+                        : @"UPDATE DocumentXml SET Xml = ?";
+                    cmd.Parameters.Add(new SQLiteParameter { Value = documentXml });
+                    cmd.ExecuteNonQuery();
+                }
+
+                transaction.Commit();
+            }
+
+            return ChangeProp(ImClone(this), im => im.DocumentXml = documentXml);
+        }
 
-                if (!string.IsNullOrEmpty(documentXml))
+        public IrtDb SetRegressionType(IrtRegressionType regressionType)
+        {
+            using (var session = OpenWriteSession())
+            using (var transaction = session.BeginTransaction())
+            {
+                if (!SqliteOperations.TableExists(session.Connection, @"DocumentXml"))
                 {
                     using (var cmd = session.Connection.CreateCommand())
                     {
-                        cmd.CommandText = !rowExists
-                            ? @"INSERT INTO DocumentXml (Xml) VALUES (?)"
-                            : @"UPDATE DocumentXml SET Xml = ?";
-                        cmd.Parameters.Add(new SQLiteParameter { Value = documentXml });
+                        cmd.CommandText = @"CREATE TABLE IF NOT EXISTS DocumentXml (Xml TEXT, RegressionType TEXT)";

```

Comment:
Can you consolidate all of the places where we ensure the "DocumentXml" table exists into one central function?
Maybe that function would have to take parameters depending on whether we needed to also ensure the "RegressionType" column exists, and whether to create the table if it does not exist, or just return false.